src/components/document-viewer/doc-fragment-creator.jsx

```diff
@@ -16,59 +18,106 @@ class DocFragmentCreator extends Component {
   constructor(props) {
     super(props);
 
+    const json = {};
+    _.forEach(this.props.schema, (value, key) => {
+      const defaultFieldValue = `// ${value.type} value`;
+      json[key] = defaultFieldValue;
+    });
+    const entryName = `New Entry - ${new Date().toString()}`;
+    const entry = {};
+    entry[entryName] = json;
+    const sourceCode = jsyaml.safeDump(entry);
+
     this.state = {
       showEditor: false,
+      lintError: null,
+      validationError: null,
+      message: '',
+      sourceCode,
     };
 
     this.saveChange = this.saveChange.bind(this);
   }
 
+  handleChange() {
+    const sourceCode = this.refs.ace.editor.getValue();
+    this.setState({ sourceCode });
+    lint(sourceCode)
+      .then(() => {
+        this.setState({ lintError: null });
+      })
+      .catch((err) => {
+        this.setState({ lintError: err.message });
+      });
+  }
+
   saveChange() {
     const source = this.refs.ace.editor.getValue();
-    DocumentService.addFragment(source);
+    const message = `Added section using Pensieve\n\n${this.state.message}`;
 
-    this.setState({
-      loading: true,
-    });
-    GitHubService.commit({ content: DocumentService.getSource() }).then(() => {
-      this.setState({
-        loading: false,
+    lint(source)
+      // When validating, strip the title field from the source
+      .then(() => schemaValidate(this.props.schema, source.replace(/^.+\n/, '')))
+      .then(() => {
+        DocumentService.addFragment(source);
+        this.setState({
+          loading: true,
+          validationError: null,
+        });
+        GitHubService.commit({ content: DocumentService.getSource(), message }).then(() => {
+          this.setState({
+            loading: false,
+          });
+          this.props.close();
+        });
+      })
+      .catch(PensieveLinterError, (err) => {
```

Comment:
Woah I didn't know you could catch by error? I usually do a switch in the `.catch`, is this a bluebird enhancement or in native promises too?